sklearn/metrics/cluster/unsupervised.py

```diff
@@ -81,6 +82,9 @@ def silhouette_score(X, labels, metric='euclidean', sample_size=None,
            <http://en.wikipedia.org/wiki/Silhouette_(clustering)>`_
 
     """
+    X = np.asarray(X)
```

Comment:
Hello @amueller ,
I add this line here because check `ndim` directly will fail in following test:
https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/metrics/cluster/tests/test_unsupervised.py#L88

This test pass list onject into `silhouette_score`, however, list object has no attribute `ndim`.